ampd/src/grpc/error.rs

```diff
@@ -115,6 +115,27 @@ impl From<&broadcaster_v2::Error> for Error {
     }
 }
 
+impl From<&cosmos::Error> for Error {
+    fn from(err: &cosmos::Error) -> Self {
+        match err {
+            cosmos::Error::GrpcConnection(_) | cosmos::Error::GrpcRequest(_) => {
+                Status::unavailable("blockchain service is temporarily unavailable")
+            }
+            cosmos::Error::QuerySmartContractState(reason) => Status::unknown(format!(
+                "failed to query smart contract state with error {reason}"
+            )),
+            cosmos::Error::GasInfoMissing
+            | cosmos::Error::AccountMissing
+            | cosmos::Error::TxResponseMissing
+            | cosmos::Error::MalformedResponse
+            | cosmos::Error::TxBuilding => {
+                Status::internal("server encountered an error processing request")
+            }
```

Comment:
why do you obfuscate all of these?

Reply:
I like having all of them here instead of using `_` so that anyone adding new errors must come here and think about the correct mapping.

Reply:
These are all internal implementations IMO that the client should not be aware of, but we can discuss if you think some of them worth exposing.

Reply:
do these get logged, too? I agree that the clients don't need to know why these failed, but for the operator of ampd it definitely is useful information

Reply:
Ya they all get logged before turning into the Status at https://github.com/axelarnetwork/axelar-amplifier/blob/26548a69b40cdfce4d988b382d3c7b10b9d21358/ampd/src/grpc/blockchain_service.rs#L90.